src/pubkey.cpp

```diff
@@ -68,8 +68,8 @@ bool CPubKey::Decompress() {
         return false;
 #ifdef USE_SECP256K1
     int clen = size();
-    int ret = secp256k1_ecdsa_pubkey_decompress((unsigned char*)begin(), &clen);
-    assert(ret);
+    if (secp256k1_ec_pubkey_decompress((unsigned char*)begin(), &clen) == 0)
+        return false;
```

Comment:
I missed this in my review before writing up my comments. This is a behavior change I meant to discuss before pushing. It doesn't factor in to the consensus call, and appears to be in error.  I would appreciate it if someone with more background would comment before I pull this out of the pull request.